lib/watchdog/watchdog_test.go

```diff
@@ -0,0 +1,172 @@
+package watchdog
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestWatchdog_StartStop tests that the watchdog starts a process, creates a PID
+// file, and stops the process when asked to.
+func TestWatchdog_StartStop(t *testing.T) {
+	t.Run("successful start and graceful stop", func(t *testing.T) {
```

Comment:
The name of the test says that here actually two tests: for start and for stop.

Reply:
done